lib/file-list.js

```diff
@@ -145,21 +71,20 @@ class FileList {
       const type = patternObject.type
 
       if (helper.isUrlAbsolute(pattern)) {
-        buckets.set(pattern, new Set([new Url(pattern, type)]))
+        this.buckets.set(pattern, [new Url(pattern, type)])
         return Promise.resolve()
       }
 
-      const mg = new Glob(pathLib.normalize(pattern), GLOB_OPTS)
+      const mg = new Glob(pathLib.normalize(pattern), { cwd: '/', follow: true, nodir: true, sync: true })
       const files = mg.found
-      buckets.set(pattern, new Set())
-
       if (_.isEmpty(files)) {
+        this.buckets.set(pattern, [])
```

Comment:
Similarly here, it looks like the implied `else` branch now does not have value assigned for `pattern` and some paths eg rejects in the promise code below would now act differently.  (I'm not against pushing the change if the tests pass but mixing with refactoring is my issue)